test/e2e/libpod_suite_test.go

```diff
@@ -0,0 +1,258 @@
+}
+
+// GrepString takes session output and behaves like grep. it returns a bool
+// if successful and an array of strings on positive matches
+func (s *PodmanSession) GrepString(term string) (bool, []string) {
+	// not implement
+	return false, nil
+}
+
+// Pull Images pulls multiple images
+func (p *PodmanTest) PullImages(images []string) error {
+	for _, i := range images {
+		p.PullImage(i)
+	}
+	return nil
+}
+
+// Pull Image a single image
+// TODO should the timeout be configurable?
+func (p *PodmanTest) PullImage(image string) error {
+	session := p.Podman([]string{"pull", image})
+	session.Wait(60)
+	Expect(session.ExitCode()).To(Equal(0))
+	return nil
+}
+
+// OutputToString formats session output to string
+func (s *PodmanSession) OutputToString() string {
+	return fmt.Sprintf("%s", s.Out.Contents())
+}
+
+// SystemExec is used to exec a system command to check its exit code or output
+func (p *PodmanSession) SystemExec(command string, args []string) (*PodmanSession, error) {
+	// not implemented yet
+	return nil, nil
+}
+
+// CreateArtifact creates a cached image in the artifact dir
+func (p *PodmanTest) CreateArtifact(image string) error {
+	imageName := fmt.Sprintf("docker://%s", image)
+	systemContext := types.SystemContext{
+		SignaturePolicyPath: p.SignaturePolicyPath,
+	}
+	policy, err := signature.DefaultPolicy(&systemContext)
+	if err != nil {
+		return errors.Errorf("error loading signature policy: %v", err)
+	}
+	policyContext, err := signature.NewPolicyContext(policy)
+	if err != nil {
+		return errors.Errorf("error loading signature policy: %v", err)
+	}
+	defer func() {
+		_ = policyContext.Destroy()
+	}()
+	options := &copy.Options{}
+
+	importRef, err := alltransports.ParseImageName(fmt.Sprintf("docker://%s", imageName))
+	if err != nil {
+		return errors.Errorf("error parsing image name %v: %v", image, err)
+	}
+
+	exportTo := filepath.Join(p.ArtifactPath, "image")
+	exportRef, err := alltransports.ParseImageName(exportTo)
+	if err != nil {
+		return errors.Errorf("error parsing image name %v: %v", exportTo, err)
+	}
+
+	return copy.Image(policyContext, exportRef, importRef, options)
+
+	return nil
+}
+
+// RestoreArtifact puts the cached image into our test store
+func (p *PodmanTest) RestoreArtifact(image string) error {
+
+	storeOptions := sstorage.DefaultStoreOptions
+	storeOptions.GraphDriverName = "vfs"
+	//storeOptions.GraphDriverOptions = storageOptions
+	storeOptions.GraphRoot = p.CrioRoot
+	storeOptions.RunRoot = p.RunRoot
+	store, err := sstorage.GetStore(storeOptions)
+
+	options := &copy.Options{}
+	if err != nil {
+		return errors.Errorf("error opening storage: %v", err)
+	}
+	defer func() {
+		_, _ = store.Shutdown(false)
+	}()
+
+	storage.Transport.SetStore(store)
+	ref, err := storage.Transport.ParseStoreReference(store, image)
+	if err != nil {
+		return errors.Errorf("error parsing image name: %v", err)
+	}
+
+	importFrom := fmt.Sprintf("dir://%s", filepath.Join(p.ArtifactPath), "image")
```

Comment:
This should be "dir:", not "dir://".  Only the "docker" transport expects a "//".

Reply:
The argument list for that `filepath.Join()` call doesn't include `image` the way the one on line 189 does?